README.md

```diff
@@ -44,8 +44,13 @@ java -jar zipkin-sparkstreaming-job.jar \
   --zipkin.sparkstreaming.sparkMaster=spark://127.0.0.1:7077
 ```
 
+
 ## Key Components
 
+The image below shows the internal architecture of zipkin spark streaming job. StreamFactory is a extensible interface that ingests data from Kafka or any other transport. The filtering step filters spans based on criteria like service name(#33). The aggregation phase groups the spans by time or trace ID. The adjuster phase is useful for making adjustments to spans that belong to the same trace. For example, the FinagleAdjuster fixes known bugs in the old finagle zipkin tracer. The final consumer stage persists the data to a storage system like ElasticSearch service. 
+
+![zipkinspark_arch](https://cloud.githubusercontent.com/assets/93836/23466070/df14857e-fe4e-11e6-89ad-2294d435b7fa.png)
```

Comment:
It might be useful to use ascii art as it is easy to change and still renders pretty nicely. It is also much easier to review, since we can make a line comment (for example questioning if we should show elasticsearch on the bottom as it might mislead people to think it only works with elasticsearch)

Another example, someone recently fixed a bug in the flow diagram without having to figure out how to render the diagram. Conversely our arch diagram rarely changes even with problems. https://github.com/openzipkin/openzipkin.github.io/pull/75

Personally, I use monodraw, though there are a number of tools for something relatively simple like this. I could also offer to convert this to an ascii variant.